disperser/common/v2/blobstore/dynamo_metadata_store.go

```diff
@@ -400,6 +387,72 @@ func (s *BlobMetadataStore) GetBlobMetadataByRequestedAt(
 	return result, lastProcessedCursor, nil
 }
 
+// queryBucketAttestation queries a single bucket for attestations within the given time range.
+func (s *BlobMetadataStore) queryBucketAttestation(ctx context.Context, bucket, startTs, endTs uint64) ([]*corev2.Attestation, error) {
+	items, err := s.dynamoDBClient.QueryIndex(
+		ctx,
+		s.tableName,
+		AttestedAtIndexName,
+		"AttestedAtBucket = :pk AND AttestedAt BETWEEN :start AND :end",
+		commondynamodb.ExpressionValues{
+			":pk":    &types.AttributeValueMemberS{Value: fmt.Sprintf("%d", bucket)},
+			":start": &types.AttributeValueMemberN{Value: strconv.FormatInt(int64(startTs), 10)},
+			":end":   &types.AttributeValueMemberN{Value: strconv.FormatInt(int64(endTs), 10)},
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("query failed for bucket %d: %w", bucket, err)
+	}
+
+	attestation := make([]*corev2.Attestation, len(items))
+	for i, item := range items {
+		attestation[i], err = UnmarshalAttestation(item)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal attestation: %w", err)
+		}
+	}
+
+	return attestation, nil
+}
+
+// GetAttestationByAttestedAt returns batches/attestations that are in AttestedAt timestamp
+// range (start, end]. Returned attestations are in ascending order of this timestamp.
+//
+// If limit > 0, returns at most that many attestations. If limit <= 0, returns all attestations in the time range.
+func (s *BlobMetadataStore) GetAttestationByAttestedAt(
+	ctx context.Context,
+	start uint64,
+	end uint64,
+	limit int,
+) ([]*corev2.Attestation, error) {
+	if start >= end {
+		return nil, errors.New("start must be less than end")
+	}
+
+	startBucket, endBucket := getAttestedAtBucketIDRange(start, end)
+
+	result := make([]*corev2.Attestation, 0)
+	for bucket := startBucket; bucket <= endBucket; bucket++ {
+		if limit > 0 && len(result) >= limit {
+			break
+		}
+
+		bucketAttestation, err := s.queryBucketAttestation(ctx, bucket, start+1, end)
```

Comment:
Do we need to do pagination? not sure how big the attestation records are / how many of them we will query at most.

Reply:
I think it'd be simpler to cap the limit (which we have yet to check how large that number should be), than pagination nested in pagination

Reply:
Dynamo can return only 1MB response at max (https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html)... This makes actually need pagination nested here

Reply:
Ok, did math: assuming 200 nonsigners, the message size will 6588 bytes, which is large; and it can return only 151 results in one query. The nonsigners are certainly a big source of cost if there are many.

There are optimizations that can be done at upper level. But at blobstore level, this should be handled as well.

```
BatchHeader: 16 bytes
AttestedAt: 8 bytes
NonSignerPubKeys: 200 * 32 = 6400 bytes (200 G1Points)
APKG2: 64 bytes (one G2Point)
QuorumAPKs: 2 * 32 = 64 bytes (values only, plus map overhead)
Sigma: 32 bytes (one signature)
QuorumNumbers: 2 * 1 = 2 bytes (assuming QuorumID is uint8)
QuorumResults: 2 * 1 = 2 bytes (values only, plus map overhead)
```